network/gossip/libp2p/peerManager_test.go

```diff
@@ -114,108 +115,128 @@ func (ts *PeerManagerTestSuite) TestUpdatePeers() {
-
+	// chooses peer interval rate deliberately long to capture on demand peer update
 	PeerUpdateInterval = time.Hour
-	err := pm.Start()
-	assert.NoError(ts.T(), err)
-
-	// wait for the first periodic update initiated after start to complete
-	assert.Eventually(ts.T(), func() bool {
-		return connector.AssertNumberOfCalls(ts.T(), "ConnectPeers", 1)
-	}, 10*time.Millisecond, 1*time.Millisecond)
 
-	// make a request for peer update
+	// creates mock connector
+	wg := &sync.WaitGroup{} // keeps track of number of calls on `ConnectPeers`
+	times, count := 2, 0    // we expect it to be called twice overall
+	wg.Add(1)               // this accounts for one invocation, the other invocation is subsequent
+	connector := new(mock.Connector)
+	// captures the first periodic update initiated after start to complete
+	connector.On("ConnectPeers", suite.ctx, testifymock.Anything).Run(func(args testifymock.Arguments) {
+		if count < times {
+			count++
+			wg.Done()
+		}
+	}).Return(nil)
+	connector.On("DisconnectPeers", suite.ctx, testifymock.Anything).Return(nil)
+	pm := NewPeerManager(suite.ctx, suite.log, idProvider, connector)
+
+	// starts peer manager and waits for first periodic update
+	require.NoError(suite.T(), pm.Start())
+	unittest.RequireReturnsBefore(suite.T(), wg.Wait, 10*time.Second,
+		"ConnectPeers is not running on startup")
+
+	// makes a request for peer update
+	wg.Add(1) // expects a call to `ConnectPeers` by requesting peer update
 	pm.RequestPeerUpdate()
 
 	// assert that a call to connect to peers is made
-	assert.Eventually(ts.T(), func() bool {
-		return connector.AssertNumberOfCalls(ts.T(), "ConnectPeers", 2)
-	}, 10*time.Millisecond, 1*time.Millisecond)
+	unittest.RequireReturnsBefore(suite.T(), wg.Wait, 1*time.Millisecond,
+		"ConnectPeers is not running on request")
 }
 
 // TestConcurrentOnDemandPeerUpdate tests that concurrent on-demand peer update request never block
-func (ts *PeerManagerTestSuite) TestConcurrentOnDemandPeerUpdate() {
+func (suite *PeerManagerTestSuite) TestConcurrentOnDemandPeerUpdate() {
 	currentIDs := unittest.IdentityListFixture(10)
 	idProvider := func() (flow.IdentityList, error) {
 		return currentIDs, nil
 	}
 
-	ctx, cancel := context.WithCancel(ts.ctx)
+	ctx, cancel := context.WithCancel(suite.ctx)
 	defer cancel()
 
 	connector := new(mock.Connector)
 	// connectPeerGate channel gates the return of the connector
 	connectPeerGate := make(chan time.Time)
 	defer close(connectPeerGate)
-	connector.On("ConnectPeers", ctx, testifymock.Anything).Return(nil).WaitUntil(connectPeerGate)
-	connector.On("DisconnectPeers", ctx, testifymock.Anything).Return(nil)
 
-	pm := NewPeerManager(ctx, ts.log, idProvider, connector)
+	// creates mock connector
+	wg := &sync.WaitGroup{} // keeps track of number of calls on `ConnectPeers`
+	// we expect it to be called twice, i.e.,
+	// one for periodic update and one for the on-demand request
+	count, times := 0, 2
```

Comment:
I think in this test we should not follow this pattern since in this we want to make sure that there were never more than 2 calls made. In this test I was trying to test that even if there are multiple concurrent `RequestPeerUpdate` calls made, only one is executed and the others are noop. If the count goes beyond 2 then it means that `RequestPeerUpdate` does not meet the expectation.
Here we can be assured that regardless of how the CI runs these routines, the blocked `ConnectPeers` return will make sure that only one call returns.

Reply:
thanks, fixed in https://github.com/onflow/flow-go/pull/99/commits/af9e5d6b5e7572b402e8721c66ac3c9355164815.